File: frontend/src/models/StatisticsDto.ts
export interface DisciplineStatDto {
  id: number
  name: string
  code: string
  totalExams: number
  countUploaded?: number
  countSegmented: number
  countCorrected: number
  avgScore: number
}

export default interface StatisticsDto {
  totalExams: number
  totalStudents: number
  countUploaded: number
  countSegmented: number
  countInDistribution: number
  countDistributed: number
  countCorrected: number
  countReleased: number
  correctionRate: number
  globalAverage: number
  totalReviews: number
  resolvedReviews: number
  reviewResolutionRate: number
  disciplineStats: DisciplineStatDto[]
}

File: backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/ems/statistics/dto/StatisticsDto.java
package pt.ulisboa.tecnico.rnl.dei.ems.statistics.dto;

import java.util.List;

public record StatisticsDto(
		long totalExams,
		long totalStudents,
		long countUploaded,
		long countSegmented,
		long countInDistribution,
		long countDistributed,
		long countCorrected,
		long countReleased,
		double correctionRate,
		double globalAverage,
		long totalReviews,
		long resolvedReviews,
		double reviewResolutionRate,
		List<DisciplineStatDto> disciplineStats) {

	public record DisciplineStatDto(
			long id,
			String name,
			String code,
			long totalExams,
			long countUploaded,
			long countSegmented,
			long countCorrected,
			double avgScore) {
	}
}
